[WIP] Update paper according to review

wdevazelhes opened the pull request:
According to revision, here are the things that need to be adressed (taken from @bellet 's mail):

- [ ] 1- Think about more general, illustrative example code
- [ ] 2- Briefly discuss positioning wrt to other libraries
- [x] 3- Highlight better the advantages of unified interface (this can be
done in relation with 2- and 3-)
- [x] 4- Clarify position wrt more recent algorithms
- [x] Talk about SCML and triplets

wdevazelhes committed:
Add triplets image

wdevazelhes committed:
add picture and gitignore

wdevazelhes committed:
first modifications

wdevazelhes committed:
Update stars, fix some pb with figure, talk about other packages, talk about triplets

wdevazelhes commented:
Hi all, I just pushed some first modifications,
I think I still need to discuss more the unified interface and the link with the matlab package (although I cited it)

Some few questions: 
- I have trouble to align well the figures with the faces... The figure of triplets is not in the middle of its legend and I don't know why
- For the example, I don't really know what to do:
  - Maybe we can keep the `fetch_pairs lfw` but remove the PCA that is maybe "ad hoc", though it will probably take longer to run (so it's bad for a user that has a small laptop but maybe on a big machine it doesn't take too much time), and it should probably be regularized if we want the algorithm to generalize ? (maybe the `gamma` can do this ?)
  - Or we can create a toy example with handcrafted points and similarities/create similarities from datasets like `make_blobs`
  - Or we can do an example with supervised learning but it doesn't illustrate well the advantage of `metric-learn`
- For citations I noticed sometimes it's quoted as "XXX and XXX", sometimes "XXX et al.", is it like when there are more than 2 authors that it uses "et al." ? Because I was thinking it could make sense to do "XXX and XXX" when it's written "authors have equal contributions" in the paper, but I don't know how to specify that

wdevazelhes committed:
Update with triplets

wdevazelhes committed:
Update pdf

terrytangyuan commented on metric-learn-jmlr.tex:
```diff
@@ -89,6 +89,7 @@ \section{Introduction}
 % (not only class supervision, unlike most packages are focused on, including another package for metric learning in Python, \texttt{PyDML})
 % , which is one interesting feature allowed by the metric learning paradigm. 
 Its API is compatible with \texttt{scikit-learn} \citep{scikit-learn}, a prominent machine learning library in Python. This allows for streamlined model selection, evaluation, and pipelining with other estimators.
+Other packages for metric-learning already exist, however, they have a different focus: \texttt{DistLearn}\footnote{\url{https://www.cs.cmu.edu/~liuy/distlearn.htm}} and \texttt{pyDML}\footnote{\url{http://github.com/jlsuarezdiaz/pydml}} contain almost only fully supervised or unsupervised metric learning algorithms, and \texttt{pytorch-metric-learning}\footnote{\url{http://github.com/KevinMusgra ...
```
```suggestion
Other packages for metric learning already exist, however, they have a different focus: \texttt{DistLearn}\footnote{\url{https://www.cs.cmu.edu/~liuy/distlearn.htm}} and \texttt{pyDML}\footnote{\url{http://github.com/jlsuarezdiaz/pydml}} contain almost only fully supervised or unsupervised metric learning algorithms, and \texttt{pytorch-metric-learning}\footnote{\url{http://github.com/KevinMusgrave/pytorch-metric-learning}} is focused on deep metric learning, using the pytorch framework.
```

terrytangyuan replied:
Also add a link to PyTorch

bellet replied:
DistLearn is not really a package, it just lists existing code in Matlab, no?
Regarding PyDML we should also mention the lack of unified interface.

wdevazelhes replied:
Will do for pytorch 
You're right for DistLearn, sorry indeed it's just a list of codes, I updated the paper

bellet replied:
@terrytangyuan maybe you want to refer to your R package also (since we mention the Matlab stuff)?
https://github.com/terrytangyuan/dml

terrytangyuan commented on metric-learn-jmlr.tex:
```diff
@@ -239,23 +245,22 @@ \section{Background on Metric Learning} \label{metriclearning}
 Detailed documentation (including installation guidelines, the description of the algorithms and the API, as well as examples) is available at \url{http://contrib.scikit-learn.org/metric-learn}.
 The development is collaborative and open to all contributors through the usual GitHub workflow of issues and pull requests.
-Community interest for the package has been demonstrated by its recent inclusion in the \texttt{scikit-learn-contrib} organization which hosts high-quality \texttt{scikit-learn}-compatible projects,\footnote{\url{https://github.com/scikit-learn-contrib/scikit-learn-contrib}} and by its more than 740 stars and 170 forks on GitHub at the time of writing.
+Community interest for the package has been demonstrated by its recent inclusion in the \texttt{scikit-learn-contrib} organization which hosts high-quality \texttt{scikit-learn}-compatible projects,\footnote{\url{https://github.com/scikit-learn-contrib/scikit-learn-contrib}} and by its more than 990 stars and 200 forks on GitHub at the time of writing.
```
There are 999 stars now so we can say 1000 here by the time we submit again!

wdevazelhes replied:
That's right ! It's already the case :) I updated the paper accordingly

terrytangyuan reviewed (commented):
Looks great to me! Just some minor comments.

bellet commented on metric-learn-jmlr.tex:
```diff
@@ -151,16 +152,20 @@ \section{Background on Metric Learning} \label{metriclearning}
+    \begin{subfigure}[t]{0.20\textwidth} % "0.45" donne ici la largeur de l'image
+        \centering \includegraphics[scale=0.35]{labels.pdf}
         \caption{class supervision}\label{fig:full}
     \end{subfigure}
-    \begin{subfigure}[t]{0.25\textwidth}
-        \centering \includegraphics[scale=0.45]{pairs.pdf}
+    \begin{subfigure}[t]{0.22\textwidth}
+        \centering \includegraphics[scale=0.35]{pairs.pdf}
         \caption{pair supervision}\label{fig:pairs}
     \end{subfigure}
-    \begin{subfigure}[t]{0.40\textwidth}
-        \centering \includegraphics[scale=0.45]{quadruplets.pdf}
+    \begin{subfigure}[t]{0.25\textwidth}
+        \centering \includegraphics[scale=0.35]{triplets.pdf}
+	\caption{triplet supervision}\label{fig:triplets}
+    \end{subfigure}
+    \begin{subfigure}[t]{0.30\textwidth}
+        \centering \includegraphics[scale=0.35]{quadruplets.pdf}
```
- remove supervision from the captions to save space
- in the triplet figure, don't use green/red color as it is not consistent with the other figures
- I think the space problem is because you have to save the triplet figure as pdf not using the whole document option but only the content so that there is no white space around the fig
- triplets.pdf should be in the images dir

bellet replied:
Actually, why not using D(A, B) < D(A, C) like in the quadruplet case (the difference being that the first image - the anchor - is the same as we can explain in the text)? It would be more consistent and would make all figs have the same height

wdevazelhes replied:
You're right! I knew I did something wrong when saving the pdf but didn't remember how to change it but now it's OK :) they are saved in the right format and it solved the alignment pb

Regarding the formatting D(A, B) < D(A, C) I was scared that it would take too much space but now that I removed the "supervision" and updated the saving format indeed it's much better with this

bellet reviewed (approved):
Here are some small comments. But I find it not very convenient to edit a paper through a detailed code review process. I'd personally prefer to take a pass after you and iterate in this fashion. So maybe just merge this after the small modifications suggested by Yuan and myself, and I can then make a pass and create a new PR for it

wdevazelhes committed:
Update metric-learn-jmlr.tex

Co-authored-by: Yuan Tang <terrytangyuan@gmail.com>

wdevazelhes committed:
Some fixes and changing the image

wdevazelhes committed:
knitpick

wdevazelhes committed:
Address final comments

wdevazelhes commented:
Hi, I addressed all the comments, I agree it's not very easy to work here from now on, feel free to merge the PR and if you want we can also continue just on overleaf, as you like

terrytangyuan commented:
@wdevazelhes Sounds good. Could you update files on Overleaf?

wdevazelhes commented:
Great, I'll merge the PR and update the overleaf

wdevazelhes commented:
Done, I updated the overleaf
